controllers/tokens/agtokens_reconciler.go

```diff
@@ -0,0 +1,132 @@
+package tokens
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	dynatracev1alpha1 "github.com/Dynatrace/dynatrace-operator/api/v1alpha1"
+	"github.com/Dynatrace/dynatrace-operator/dtclient"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	SecretsName = "dynatrace-tokens"
```

Comment:
In case multiple environments are configured the token would get overwritten by each reconciliation. Therefore the name of the secret must contain the name of the DynaKube object.

Reply:
multiple environments == multiple operators? Secret are held in operator's namespace. 
Do we support running multiple instances of `dynatrace-operator` in single namespace?

Reply:
note:
same logic is used for https://github.com/Dynatrace/dynatrace-operator/blob/master/api/v1alpha1/dynakube_types.go#L24

Reply:
related discussion: https://github.com/Dynatrace/dynatrace-operator/pull/167#discussion_r659588372

Reply:
> multiple environments == multiple operators? Secret are held in operator's namespace.
> Do we support running multiple instances of `dynatrace-operator` in single namespace?

Single operator handles multiple DynaKube objects. Each of these DynaKubes can belong to different environments. Only one namespace involved.